frontend/src/components/WaitingRoom/UserNameInput/UserNameInput.tsx

```diff
@@ -67,7 +68,10 @@ const UsernameInput = ({ username, setUsername }: UserNameInputProps): ReactElem
   const handleJoinClick = (event: MouseEvent) => {
     event.preventDefault();
     if (validateForm() && roomName) {
-      const sanitizedRoomName = encodeURIComponent(roomName);
+      const validatedRoomName = validateRoomName(roomName);
+      if (!validatedRoomName) {
+        return;
+      }
```

Comment:
Nit: probably more readable with some light refactoring.

```
if (!isValidRoomName(roomName)) {
  return;
}
```

Reply:
Does `encodeURIComponent` have any value?  I think it might be used elsewhere, yes?